pkg/controller/framework/controller.go

```diff
@@ -68,6 +68,11 @@ type Controller struct {
 	reflectorMutex sync.RWMutex
 }
 
+type ControllerInterface interface {
```

Comment:
I just did some quick scanning of the code, and it seems that Controller itself has just one more additional method "Requeue", and it seems it is not used anywhere.

So can you please add a TODO, to make the "Controller" private, and convert all references to use ControllerInterface instead?